src/util_gen.cpp

```diff
@@ -470,6 +477,14 @@ bool Util::isValidFd(int fd)
   return _real_fcntl(fd, F_GETFL, 0) != -1;
 }
 
+// On failure, returns 0, with errno == EFAULT
+bool Util::isReadableMemoryAddress(const char *addr) {
+  // FIXME: We wanted to call libc:open(), but _real_open goes to second wrapper
+  int fd = _real_syscall(SYS_open, "/dev/null", O_WRONLY);
+  int rc = write(fd, addr, 1);
```

Comment:
I looked it up, and I'm getting a different reading of the POSIX standard.  I'm seeing:
    _Implementations are allowed, but not required, to perform error checking for write() requests of zero bytes._
at:  http://pubs.opengroup.org/onlinepubs/009695399/functions/write.html
This seems to say that a write of zero bytes from a bad memory address can succeed.